### ISSUE TITLE
Add support for headers parsing in csv function.

### ISSUE THREAD

**Issue by Midbin** [typst/typst#2618]
### Description

It would be useful if the `#csv` function could have an option to parse headers. If the headers option is enabled, the function would parse the first line of the `.csv` as a header row. The Output would be an array of dictionarys, where the header-label would be the key and the value the value.

For example when using:
```
#csv("zoo.csv", has-headers: true)
```
for `zoo.csv`:
```
name,species,weight,length
Debby,Rhinoceros,1900kg,390cm
Fluffy,Tiger,115kg,310cm
Sleepy,Dolphin,150kg,180cm
```
would become:
```typ
(
  (
    name: "Debby",
    species: "Rhinoceros",
    weight: "1900kg",
    length: "390cm",
  ),
  (
    name: "Fluffy",
    species: "Tiger",
    weight: "115kg",
    length: "310cm",
  ),
  (
    name: "Sleepy",
    species: "Dolphin",
    weight: "150kg",
    length: "180cm",
  ),
)
```

has-headers would be disabled by default, which ensures that no existing code would change.
This would be an additive change.

### Use Case

This becomes very useful when using `.csv`files with a lot of rows. I had to plot (using cetz) a csv file using 18 rows and counting them by hand becomes tedious and error prone. For example I had an off by one error, wich meant that I did not select `wind-offshore`, but `wind-onshore`. This kind of error may not be easily detected based on the data (as in my example).

With this change
```
plot.add(productionData.map(x => x.at(0), x.at(15)))
```
would become
```typ
plot.add(productionData.map(x => x.time, x.wind-offshore))
```

**Comment by sitandr** [comment]
I'm totally not sure this is a good idea.

The thing you want is a flag that will magically transform your data. Transform in a very obscure way. If you use the same logic, you would need a flag to output data in a form of `(heading1: (value1, value2, value3), heading2: (value1, value2, value3))` and so on. And do that not only for `csv` reader, but also many others.

The fact is that converting from basic format with headings to yours may be done with few Typst code lines. I don't see why that should be inconsistently built-in.

I want to add that at least now Typst in general is not very good suited for working with data. Using `python` with `pandas` or `polars` would be a much better choice there. I don't think it is a good idea to clone all their functionality into Typst. Typst isn't about easily working with data.

**Comment by Midbin** [comment]
Thanks for your feedback. 

While a agree that Typst is not a good language to do data processing, I think it should still have good data display support. For example the dataset I wanted to display is build with another complex tool, but graphs are usually generated using a typesetter, for example `Latex`with `pgfplots`. That workflow is, in my opinion, something that Typst should also try to cover. 

I also want to state that header rows are a pretty common csv feature. While a universally agreed upon csv specification does not exist, header rows are used quite often. The [rfc 4814](https://datatracker.ietf.org/doc/html/rfc4180) does mention header rows and the rust csv libary (which typst uses) has csv headers [enabled by default](https://docs.rs/csv/latest/csv/struct.ReaderBuilder.html#method.has_headers) (typst does manually disable them).

I would also not argue that this will magically transform the data. This specific transformation is very common, for example the rust csv libary has a whole section on [reading headers](https://docs.rs/csv/latest/csv/tutorial/index.html#reading-headers) and reading csv rows [into a hashmap or using csv headers to fill a record using serde](https://docs.rs/csv/latest/csv/tutorial/index.html#reading-with-serde) in its [basic tutorial](https://docs.rs/csv/latest/csv/tutorial/index.html). The [python csv reader](https://docs.python.org/3/library/csv.html) also has a class DictReader which `maps the but maps the information in each row to a [dict](https://docs.python.org/3/library/stdtypes.html#dict)`whose keys are by default the header fields (you can also specify other names). The last and for typst maybe most important reference is latex where pretty much all csv readers (that I know of) do this kind of transformation. Take for example [csvtools](https://ctan.org/pkg/csvtools), [which generates](https://mirror.physik.tu-berlin.de/pub/CTAN/obsolete/macros/latex/contrib/csvtools/doc/csvtools.pdf) `\insert... commands (...) to access orresponding elements in the row currently being processed. For example, suppose the first line of the CSV file looks like: Name,Address,Time,Date then the commands \insertName, \insertAddress, \insertTime and \insertDate are created`. Or for example [pdfplots](https://www.ctan.org/pkg/pgfplots), which uses the header fields to specify which datapoints you want to use out of the dataset, and for which axis.

In conclusion I do not thing this transofmation is _magical_ or _very obscure_. I would argue that this is a very common and expected way to parse csv data and some variation of this transformation is present in pretty much every csv parsing library I ever used (there are more than quoted here for example java, swift,...) and often the default (rust, swift). 

Please also note that implementing this into typst is pretty easy since the rust csv libary typst uses alreay supports this.

**Comment by sitandr** [comment]
Okay, these are really good arguments, I think I agree with you now.

Thanks for explaining!
